### DISCUSSION TITLE
Problems/Questions using .NET MAUI App (Dan Siegel) template Visual Studio 2022

### DISCUSSION THREAD

**Discussion by nk-alex** [PrismLibrary/Prism#3104] [.NET MAUI]
I've just started the transition from Xamarin to MAUI due to Xamarin's EOL. One of the things I've done first is to search for Prism template for my Windows 11 Visual Studio 2022 (17.9.2):

`dotnet new --install Prism.Templates::8.1.97 --force`

Which prompts:

```
Success: Prism.Templates::8.1.97 installed the following templates:
Template Name                    Short Name       Language  Tags                                                        
-------------------------------  ---------------  --------  ------------------------------------------------------------
Prism .NET MAUI App              prism-maui       [C#]      MAUI/Android/iOS/macOS/Mac Catalyst/Windows/Tizen           
Prism Blank App (Uno Platform)   uno-blank        [C#]      Prism/Xamarin/Uno Platform/WebAssembly/iOS/Android/WinUI/UWP
Prism Blank App (WPF)            wpf-core-blank   [C#]      Desktop                                                     
Prism Blank App (Xamarin.Forms)  xf-blank         [C#]      Prism/Xamarin/Xamarin.Forms                                 
Prism Full App (WPF)             wpf-core-full    [C#]      Desktop                                                     
Prism Module (WPF)               wpf-module-core  [C#]      Desktop                                                     
Prism Module (Xamarin)           xf-module        [C#]      Prism/Xamarin/Xamarin.Forms       
```                          

Now I restart my Visual Studio and I select "Create a new project". Now I follow the steps to create a new project using .NET MAUI App (Dan Siegel) template.

Visual Studio now gets stuck on "Creating Project..." and finally crashes after a minute stuck. When I check the selected path, seems it has created a project, but I'm not sure it's usable:

![image](https://github.com/PrismLibrary/Prism/assets/101177816/cc499425-c1bd-43b9-92f8-e57089d31bc9)

Now I just delete this project and try to create it a different way. Everytime I follow the steps described above I get the same result.
This second time, I open my Visual Studio and instead of pressing "Create a new project", I select "Continue without code". Now I go File > New > Project > Prims .NET MAUI App (Dan Siegel) > Create (at same path as before) and now works as expected. Visual Studio is not crashing and this is how the create project is looking:

![image](https://github.com/PrismLibrary/Prism/assets/101177816/ae91405f-d249-4a40-bbed-9ca7782179ae)

Now what I find is the following scenario:

![image](https://github.com/PrismLibrary/Prism/assets/101177816/bbfe2d2c-6e85-4708-9f8d-e8102b9d6cbb)

Seems like the project has been created for .NET6.0 and no emulators are showing. Also the Solution Explorer is showing empty.
So I just close Visual Studio and I restart it from double click the project. Now Solution Explorer is working as expected but I can't see any emulator but "Windows Machine":

![image](https://github.com/PrismLibrary/Prism/assets/101177816/1ed0e25a-803c-4dd1-ab34-3840aeda76ab)

Also a bunch of compilation errors related to net6.0 (this is my fault since I don't have it installed):

![image](https://github.com/PrismLibrary/Prism/assets/101177816/4978960c-b476-41ae-8314-f23b4b11043f)

To be able to get rid of this errors, I just double click the solution and replace all "net6.0" instances by "net8.0" from project configuration file. Now I can compile the project with no problems. But still no emulators showing for the different platforms. Also I can't deploy it using "Windows Machine":

![image](https://github.com/PrismLibrary/Prism/assets/101177816/2a1c0c1c-38be-4385-9a69-184f1cdc845b)

If I create the project using the normal .NET MAUI App project template I can see and deploy on my emulators with no problem:

![image](https://github.com/PrismLibrary/Prism/assets/101177816/68b8b92c-0895-4bbb-a62b-907bd2bd0f8d)

To solve this issue, in my case, I had to replace the SupportedOSPlarformVersions in my project configuration file:

### Original one. Can't deploy and no emulators showing:
```
<SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'ios'">14.2</SupportedOSPlatformVersion>
    <SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'maccatalyst'">14.0</SupportedOSPlatformVersion>
    <SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'android'">21.0</SupportedOSPlatformVersion>
    <SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'">10.0.17763.0</SupportedOSPlatformVersion>
    <TargetPlatformMinVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'">10.0.17763.0</TargetPlatformMinVersion>
    <SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'tizen'">6.5</SupportedOSPlatformVersion>
```

### New one. I can deploy and all emulators showing:
```
		<SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'ios'">11.0</SupportedOSPlatformVersion>
		<SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'maccatalyst'">13.1</SupportedOSPlatformVersion>
		<SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'android'">21.0</SupportedOSPlatformVersion>
		<SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'">10.0.17763.0</SupportedOSPlatformVersion>
		<TargetPlatformMinVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'">10.0.17763.0</TargetPlatformMinVersion>
		<SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'tizen'">6.5</SupportedOSPlatformVersion>
```

![image](https://github.com/PrismLibrary/Prism/assets/101177816/df79b472-52be-47e2-876b-70b8cea5292a)

I really don't understand why ios and maccatalyst versions are making emulators not showing. I guess this is a Visual Studio thing.

## Questions
1. Is anything I'm doing wrong? If I'm not doing anythig wrong, I can go over this process again and deliver all the logs you need. I will be happy to help.
2. Is there any way of selecting .NET version from all the installed before creating project? Something like the normal .NET MAUI App template. In my case the project is created by default with a .NET version I don't even have installed.

**Comment by egz019** [comment]
You're not alone dude. I bump into this too and trying to figure out in the community about this.

**Comment by wj-wong** [comment]
Thanks!
Your answer solved my problem, it seem like the templates have been moved to a private repo and are currently only available to those with a Commercial Plus license. (https://github.com/PrismLibrary/Prism/issues/2618#issuecomment-2097018672)
